components/pkg-export-container/src/cli.rs

```diff
@@ -1,357 +1,288 @@
+                                                           .value_name("IMAGE_NAME")
+                                                           .help("Image name (default: \
+                                                                  {{pkg_origin}}/{{pkg_name}} \
+                                                                  supports: {{pkg_origin}}, \
+                                                                  {{pkg_name}}, {{pkg_version}}, \
+                                                                  {{pkg_release}}, {{channel}})"));
 
-#[derive(Clone)]
-struct Cli<'a, 'b>
-    where 'a: 'b
-{
-    pub app: App<'a, 'b>,
-}
+    let cmd = add_base_packages_args(cmd);
+    let cmd = add_builder_args(cmd);
+    let cmd = add_tagging_args(cmd);
+    let cmd = add_publishing_args(cmd);
+    let cmd = add_memory_arg(cmd);
+    let cmd = add_layer_arg(cmd);
+    let cmd = add_pkg_ident_arg(cmd);
+    let cmd = add_engine_arg(cmd);
 
-impl<'a, 'b> Cli<'a, 'b> {
-    fn new(name: &str, about: &'a str) -> Self {
-        Cli { app: clap_app!(
-              (name) =>
-              (about: about)
-              (version: VERSION)
-              (author: "\nAuthors: The Habitat Maintainers <humans@habitat.sh>\n\n")
-              (@arg IMAGE_NAME: --("image-name") -i +takes_value
-                  "Image name (default: \"{{pkg_origin}}/{{pkg_name}}\" supports: \
-                   {{pkg_origin}}, {{pkg_name}}, {{pkg_version}}, {{pkg_release}}, {{channel}})")
-              ), }
+    if cfg!(windows) {
+        add_base_image_arg(cmd)
+    } else {
+        cmd
     }
+}
 
-    fn add_base_packages_args(self) -> Self {
-        let app = self
-            .app
-            .arg(
-                Arg::with_name("HAB_PKG")
-                    .long("hab-pkg")
-                    .value_name("HAB_PKG")
-                    .validator(valid_ident_or_hart)
-                    .help(
-                        "Habitat CLI package identifier (ex: acme/redis) or filepath to a Habitat \
+fn add_base_packages_args(cmd: Command) -> Command {
+    cmd.arg(
+        Arg::new("HAB_PKG")
+            .long("hab-pkg")
+            .value_name("HAB_PKG")
+            .default_value(super::DEFAULT_HAB_IDENT)
+            .value_parser(HabHartIdParser)
+            .help(
+                "Habitat CLI package identifier (ex: acme/redis) or filepath to a Habitat \
                          artifact (ex: /home/acme-redis-3.0.7-21120102031201-x86_64-linux.hart) \
                          to install (default: core/hab)",
-                    ),
-            )
-            .arg(
-                Arg::with_name("HAB_LAUNCHER_PKG")
-                    .long("launcher-pkg")
-                    .value_name("HAB_LAUNCHER_PKG")
-                    .validator(valid_ident_or_hart)
-                    .help(
-                        "Launcher package identifier (ex: core/hab-launcher) or filepath to a \
+            ),
```

Comment:
We should be able to remove the "(default: ...)" from the help text since clap outputs that for us.